### PR TITLE
NP-4738: Added TokenExpiredException and TokenInvalidException

### PR DESCRIPTION
**Pull request by adamfox-nanopay**

### CODE REVIEW

File: src/foam/nanos/auth/token/TokenExpiredException.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  name: 'TokenExpiredException',
+  package: 'foam.nanos.auth.token',
+  extends: 'foam.nanos.auth.AuthenticationException',
+  javaGenerateDefaultConstructor: false,
+  javaGenerateConvenienceConstructor: false,
+
+  properties: [
+    {
+      name: 'exceptionMessage',
+      value: 'Token has expired'
```
Comment (jlhughes) [inline src/foam/nanos/auth/token/TokenExpiredException.js]: This message isn't needed. The foam exception system will parse the exception name into message:  Token Expired
and do translation on this if required.

File: src/foam/nanos/auth/token/TokenInvalidException.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  name: 'TokenInvalidException',
+  package: 'foam.nanos.auth.token',
+  extends: 'foam.nanos.auth.AuthenticationException',
+  javaGenerateDefaultConstructor: false,
+  javaGenerateConvenienceConstructor: false,
+
+  properties: [
+    {
+      name: 'exceptionMessage',
+      value: 'Token not found'
```
Comment (jlhughes) [inline src/foam/nanos/auth/token/TokenInvalidException.js]: Similar here.  
options are to rename to InvalidToken, so the english text becomes 
Invalid Token.

File: src/foam/nanos/auth/token/TokenExpiredException.js (hunk 1 above)
Comment (adamfox-nanopay) [reply src/foam/nanos/auth/token/TokenExpiredException.js]: These exceptions are handled in Controller.js and use the message to determine error 
e.g.
 `           if ( e.message && e.message === "Token has expired" ) {`
Is there another way to do this on client side without exceptionMessage property?

Comment (adamfox-nanopay) [reply src/foam/nanos/auth/token/TokenExpiredException.js]: AND if i remove the exceptionMessage property e.message is empty

Comment (jlhughes) [reply src/foam/nanos/auth/token/TokenExpiredException.js]: Recommend you use id which is (cls_.id) and in this case becomes: "TokenExpiredException".

**Review by lchanmann — APPROVED** [review]

**Pull request merged**